File: frameworks/java/quarkus/UnitTests/src/test/java/unittests/StaticTests.java
package unittests;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static unittests.Http.get;

import io.quarkus.test.junit.QuarkusTest;
import io.restassured.response.Response;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

@QuarkusTest
class StaticTests {

    // rb:test static.file
    @Test
    @Tag("static.file")
    void theFileIsSentAsItIsWithItsLengthAndAge() {
        Response response = get("/static/items.large.json");

        byte[] file = Expected.bytes("items.large.json");
        assertArrayEquals(file, response.asByteArray());
        assertEquals("application/json", response.contentType());
        assertEquals(String.valueOf(file.length), response.header("content-length"));
        assertNotNull(response.header("last-modified"));
    }

    @Test
    void aFileThatIsNotThereIsA404() {
        assertEquals(404, get("/static/nothing.json").statusCode());
    }
}

File: frameworks/java/ktor/UnitTests/src/test/kotlin/unittests/StaticTests.kt
package unittests

import io.ktor.client.request.get
import io.ktor.client.statement.bodyAsBytes
import io.ktor.http.ContentType
import io.ktor.http.HttpHeaders
import io.ktor.http.HttpStatusCode
import io.ktor.http.contentType
import kotlin.test.assertContentEquals
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test

class StaticTests {
    // rb:test static.file
    @Test
    @Tag("static.file")
    fun `the committed file is served byte for byte with its last modified`() = corpusTest {
        val file = Expected.raw("items.large.json")

        val response = client.get("/static/items.large.json")

        assertEquals(HttpStatusCode.OK, response.status)
        assertEquals(ContentType.Application.Json, response.contentType()?.withoutParameters())
        assertEquals(file.size.toString(), response.headers[HttpHeaders.ContentLength])
        assertNotNull(response.headers[HttpHeaders.LastModified])
        assertContentEquals(file, response.bodyAsBytes())
    }
}
